src/main/java/cn/nukkit/Player.java

```diff
@@ -1314,7 +1314,7 @@ public boolean isSpectator() {
     }
 
     public boolean isAdventure() {
-        return (this.gamemode & 0x02) > 0;
+        return (this.gamemode & 0x02) > 0 && !this.isSpectator();
```

Comment:
Why are there even bit-wise operators here?

Reply:
Idk. I think we can judge directly without using bit-wise operators.
```java
    public boolean isSurvival() {
        return this.gamemode == 0;
    }

    public boolean isCreative() {
        return this.gamemode == 1;
    }

    public boolean isSpectator() {
        return this.gamemode == 3;
    }

    public boolean isAdventure() {
        return this.gamemode == 2;
    }
```

Reply:
Yes, this is what I would have expected in the first place. Also, we could use the constants defined in player like `Player.ADVENTURE` so we don't have magic numbers all over the place.